helper/helper/src/main/java/com/linkedin/avroutil1/normalization/AvroUtilSchemaNormalization.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2023 LinkedIn Corp.
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+
+package com.linkedin.avroutil1.normalization;
+
+import com.linkedin.avroutil1.compatibility.AvroCompatibilityHelper;
+import com.linkedin.avroutil1.compatibility.AvroVersion;
+import com.linkedin.avroutil1.compatibility.AvscGenerationConfig;
+import com.linkedin.avroutil1.compatibility.AvscWriter;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.avro.Schema;
+
+
+public class AvroUtilSchemaNormalization {
+
+  public enum FingerprintingAlgo {
+    CRC_64 ("CRC-64-AVRO"),
+    MD5_128 ("MD5"),
+    SHA_256 ("SHA-256");
+
+    private final String name;
+    FingerprintingAlgo(String fpName) {
+      name = fpName;
+    }
+
+    public String toString() { return name; }
+  }
+
+  /**
+   * Returns "Parsing Canonical Form" of a schema as defined by Avro spec.
+   * @param schema an avro schema
+   * @param config generation config
+   * @param schemaPlugins List of AvscWriterPlugins to be applied on the schema during avsc generation
+   * @return parsing canonical form of the schema
+   */
+  public static String getCanonicalForm(Schema schema, AvscGenerationConfig config, List<AvscWriterPlugin> schemaPlugins) {
+    if (AvroCompatibilityHelper.getRuntimeAvroCompilerVersion().laterThan(AvroVersion.AVRO_1_4)) {
+      AvscWriter writer = AvroCompatibilityHelper.getAvscWriter(config, schemaPlugins);
+      return writer.toAvsc(schema);
+    }
+    else {
+      throw new UnsupportedOperationException("Canonicalize need avro > 1.4");
+    }
+  }
+
+  /**
+   * Returns {@link #fingerprint} applied to the parsing canonical form of the
+   * supplied schema.
+   * @param fpName fingerprint algorithm name
+   * @param s schema who's parsing canonical form will be fingerprinted
+   * @return fingerprint of given schema's parsing canonical form
+   * @throws NoSuchAlgorithmException if algorithm if not found, shouldn't happen
+   */
+  public static byte[] parsingFingerprint(FingerprintingAlgo fpName, Schema s, AvscGenerationConfig config,
```

Comment:
Can we make this an active name? i.e. `generateFingerprint()` or `canonicalizeSchema()`

Also,